test/dnnl/DnnlTestUtils.h

```diff
@@ -21,7 +21,21 @@
 #include <sstream>
 #include <string>
 
-static std::string read_str_resource(const std::string &name) {
+#if __cplusplus > 202002L
+#include <stdfloat>
+#else
+namespace std {
+#if defined(__SIZEOF_FLOAT__) && __SIZEOF_FLOAT__ == 4
+using float32_t = float;
+#elif defined(__SIZEOF_DOUBLE__) && __SIZEOF_DOUBLE__ == 4
+using float32_t = double;
+#else
+static_assert(false, "No 32-bit floating point type available");
+#endif
+} // namespace std
+#endif
```

Comment:
It should be some header in ProjectRoot/include, because the API tests do not have access to the internal headers. I'm not sure if it currently makes sense to create a new public header just for a single definition, but it could be moved later, when there is one.